src/rust/ide/lib/view/src/graph_editor/component/breadcrumbs/breadcrumb.rs

```diff
@@ -0,0 +1,413 @@
+//! This module provides a clickable view for a single breadcrumb.
+
+use crate::prelude::*;
+
+use super::GLYPH_WIDTH;
+use super::HORIZONTAL_MARGIN;
+use super::VERTICAL_MARGIN;
+use super::TEXT_SIZE;
+
+use enso_frp as frp;
+use ensogl::animation::linear_interpolation;
+use ensogl::data::color;
+use ensogl::display;
+use ensogl::display::Attribute;
+use ensogl::display::Buffer;
+use ensogl::display::object::ObjectOps;
+use ensogl::display::scene::Scene;
+use ensogl::display::shape::*;
+use ensogl::display::shape::text::glyph::system::Line;
+use ensogl::display::shape::text::glyph::system::GlyphSystem;
+use ensogl::display::Sprite;
+use ensogl::gui::component;
+use ensogl::gui::component::Animation;
+use enso_protocol::language_server::MethodPointer;
+use logger::enabled::Logger;
+use logger::AnyLogger;
+use nalgebra::Vector2;
+use std::f32::consts::PI;
+
+
+
+// =================
+// === Constants ===
+// =================
+
+const ICON_MARGIN          : f32 = HORIZONTAL_MARGIN;
+const ICON_RADIUS          : f32 = 8.0;
+const ICON_SIZE            : f32 = ICON_RADIUS * 2.0;
+const ICON_RING_WIDTH      : f32 = 2.0;
+const ICON_ARROW_SIZE      : f32 = 5.0;
+const SEPARATOR_SIZE       : f32 = 8.0;
+const SEPARATOR_LINE_WIDTH : f32 = 3.0;
+const SEPARATOR_MARGIN     : f32 = HORIZONTAL_MARGIN;
+const FULL_COLOR           : color::Rgba = color::Rgba::new(1.0, 1.0, 1.0, 0.7);
+const TRANSPARENT_COLOR    : color::Rgba = color::Rgba::new(1.0, 1.0, 1.0, 0.4);
+
+
+
+// ==================
+// === Background ===
+// ==================
+
+mod background {
+    use super::*;
+
+    ensogl::define_shape_system! {
+        () {
+            let bg_color = color::Rgba::new(0.0,0.0,0.0,0.000_001);
+            Plane().fill(bg_color).into()
+        }
+    }
+}
+
+
+
+// ============
+// === Icon ===
+// ============
+
+mod icon {
+    use super::*;
+
+    ensogl::define_shape_system! {
+        (opacity:f32) {
+                let outer_circle  = Circle((ICON_RADIUS - 0.5).px());
+                let inner_circle  = Circle((ICON_RADIUS - 0.5 - ICON_RING_WIDTH).px());
+                let ring          = outer_circle - inner_circle;
+                let size          = ICON_ARROW_SIZE;
+                let arrow         = Triangle(size.px(),size.px()).rotate((PI/2.0).radians());
+                let arrow         = arrow.translate_x(0.5.px());
+                let shape         = ring + arrow;
+                let full_color    = format!("vec4({},{},{},{})"
+                    ,FULL_COLOR.red,FULL_COLOR.green,FULL_COLOR.blue,FULL_COLOR.alpha);
+                let transparent_color = format!("vec4({},{},{},{})"
+                    ,TRANSPARENT_COLOR.red,TRANSPARENT_COLOR.green,TRANSPARENT_COLOR.blue
+                    ,TRANSPARENT_COLOR.alpha);
+                let color = format!("mix({},{},{})",transparent_color,full_color,opacity);
```

Comment:
ehh, we need to expose better way to do it. Im happy you found a way around!

Reply:
Yeah, it's damn hacky.